src/Diagnostics.ModelsAndUtils/Models/ResponseExtensions/DetectorStatus.cs

```diff
@@ -21,13 +23,48 @@ public static class ResponseDetectorStatusExtensions
         /// }
         /// </code>
         /// </example>
-        public static void SetDetectorStatus(this Response response, DetectorStatus health, string message = null)
+        public static void SetDetectorStatus(this Response response, DetectorStatus status, string message = null)
         {
             response.Status = new Status()
             {
-                StatusId = health,
+                StatusId = status,
                 Message = message
             };
         }
+
+        public static void UpdateDetectorStatusFromInsights(this Response response)
+        {
+
+            if (response.Status != null)
+            {
+                return;
+            }
+            
+            var status = response.Dataset
+                .Where(set => set.RenderingProperties.Type == RenderingType.Insights || set.RenderingProperties.Type == RenderingType.DynamicInsight)
+                .Select(set =>
+                {
+                    if (set.RenderingProperties.Type == RenderingType.DynamicInsight)
+                    {
+                        return (int)((DynamicInsightRendering)set.RenderingProperties).Status;
+                    }
+                    else
+                    {
+                        int lowestStatus = (int)InsightStatus.None;
+                        foreach (DataRow row in set.Table.Rows)
+                        {
+                            var insightStatusInt = (int)Enum.Parse(typeof(InsightStatus), row["Status"].ToString());
+                            if (lowestStatus > insightStatusInt)
+                            {
+                                lowestStatus = insightStatusInt;
+                            }
+                        }
+                        return lowestStatus;
+                    }
+                    
+                }).OrderBy(st => st).FirstOrDefault();
```

Comment:
Since you reordered the enum values for InsightStatus then this Ordery and FirstOrDefault should return back the most critical status (no pun intended) of the insights right?

Reply:
That is the intent yes, the status of the detector will be most critical of its insights.